pkg/newstorage/newstorage.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package newstorage
+
+import "errors"
+
+// ErrStoreNotFound is returned when a store is not found.
+var ErrStoreNotFound = errors.New("store not found")
+
+// ErrDataNotFound is returned when data is not found.
+var ErrDataNotFound = errors.New("data not found")
+
+// StoreConfiguration represents the configuration of a store.
+type StoreConfiguration struct {
+	TagNames []string
+}
+
+// QueryOptions represents various options for Query calls in a store.
+type QueryOptions struct {
+	pageSize int
+}
+
+// QueryOption represents an option for a Query call in a store.
+type QueryOption func(opts *QueryOptions)
+
+// WithPageSize sets the maximum page size for data retrievals done within the Iterator returned by the Query call.
+// Paging is handled internally by the Iterator. Higher values may reduce CPU time and the number of database calls at
+// the expense of higher memory usage.
+func WithPageSize(size int) QueryOption {
+	return func(opts *QueryOptions) {
+		opts.pageSize = size
+	}
+}
+
+// Tag represents a Name + Value pair that can be associated with a k,v pair for querying later.
+type Tag struct {
+	Name  string
+	Value string
```

Comment:
The rationale behind `value` should be documented.

Reply:
Done in my new PR: https://github.com/hyperledger/aries-framework-go/pull/2426/files#diff-98db716f6d6bedd511a5f903dc6d785b6f60df130787f50612d64804a1f8195fR49